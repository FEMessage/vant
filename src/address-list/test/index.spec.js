import { mount } from '../../../test';
import AddressList from '..';

const list = [
  {
    id: '1',
    name: '张三',
    tel: '13000000000',
    address: '浙江省杭州市西湖区文三路 138 号东方通信大厦 7 楼 501 室',
    isDefault: true,
  },
  {
    id: '2',
    name: '李四',
    tel: '1310000000',
    address: '浙江省杭州市拱墅区莫干山路 50 号',
  },
];

test('unswitchable', () => {
  const wrapper = mount(AddressList, {
    propsData: {
      list,
      switchable: false,
    },
  });

  expect(wrapper).toMatchSnapshot();
});

test('set-default event', () => {
  const onSetDefault = jest.fn();
  const wrapper = mount(AddressList, {
    propsData: {
      list,
    },
    context: {
      on: {
        'set-default': onSetDefault,
      },
    },
  });

  wrapper.find('.van-radio__icon').trigger('click');

  expect(onSetDefault).toHaveBeenCalledTimes(1);
});

test('click-item event', () => {
  const onClickItem = jest.fn();
  const wrapper = mount(AddressList, {
    propsData: {
      list,
    },
    context: {
      on: {
        'click-item': onClickItem,
      },
    },
  });

  wrapper.find('.van-address-item__content').trigger('click');

  expect(onClickItem).toHaveBeenCalledTimes(1);
});
