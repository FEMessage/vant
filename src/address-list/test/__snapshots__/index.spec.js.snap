// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unswitchable 1`] = `
<div class="van-address-list">
  <div role="radiogroup" class="van-radio-group">
    <div role="button" tabindex="0" class="van-cell van-cell--clickable van-address-item">
      <div class="van-cell__value van-cell__value--alone">
        <div class="van-address-item__content">
          <div class="van-address-item__name">张三，13000000000</div>
          <div class="van-address-item__address">浙江省杭州市西湖区文三路 138 号东方通信大厦 7 楼 501 室</div>
        </div>
        <div class="van-address-item__bar">
          <div role="radio" tabindex="-1" aria-checked="false" class="van-radio van-address-item__set-default">
            <div class="van-radio__icon van-radio__icon--round"><i class="van-icon van-icon-success">
                <!----></i></div><span class="van-radio__label">设为默认</span>
          </div>
          <div class="van-address-item__icons-wrapper"><span class="van-address-item__edit" style="display: flex;"><i class="van-icon van-icon-edit"><!----></i></span><span class="van-address-item__delete" style="display: flex;"><i class="van-icon van-icon-delete"><!----></i></span></div>
        </div>
      </div>
    </div>
    <div role="button" tabindex="0" class="van-cell van-cell--clickable van-address-item">
      <div class="van-cell__value van-cell__value--alone">
        <div class="van-address-item__content">
          <div class="van-address-item__name">李四，1310000000</div>
          <div class="van-address-item__address">浙江省杭州市拱墅区莫干山路 50 号</div>
        </div>
        <div class="van-address-item__bar">
          <div role="radio" tabindex="-1" aria-checked="false" class="van-radio van-address-item__set-default">
            <div class="van-radio__icon van-radio__icon--round"><i class="van-icon van-icon-success">
                <!----></i></div><span class="van-radio__label">设为默认</span>
          </div>
          <div class="van-address-item__icons-wrapper"><span class="van-address-item__edit" style="display: flex;"><i class="van-icon van-icon-edit"><!----></i></span><span class="van-address-item__delete" style="display: flex;"><i class="van-icon van-icon-delete"><!----></i></span></div>
        </div>
      </div>
    </div>
  </div><button class="van-button van-button--danger van-button--large van-button--square van-address-list__add"><span class="van-button__text">新增地址</span></button>
</div>
`;
